Assets/VRTK/Scripts/VRTK_RealMovement.cs

```diff
@@ -0,0 +1,350 @@
+    /// <remarks>
+    /// It is recommend you combine this with the VRTK_HeightAdjustTeleport
+    /// </remarks>
+    /// <example>
+    /// `VRTK/Examples/042_CameraRig_RealMovement` shows how the user can move and traverse colliders.
+    /// </example>
+    public class VRTK_RealMovement : MonoBehaviour
+    {
+
+        /// <summary>
+        /// Options for testing if a play space fall is valid
+        /// </summary>
+        /// <param name="HeadsetAndControllers">Track both headset and controllers for movement calculations.</param>
+        /// <param name="ControllersOnly">Track only the controllers for movement calculations.</param>
+        /// <param name="HeadsetOnly">Track only headset for movement caluclations.</param>
+        public enum ControlOptions
+        {
+            HeadsetAndControllers,
+            ControllersOnly,
+            HeadsetOnly,
+        }
+        
+        public enum DirectionalMethod
+        {
+            Gaze,
+            DumbDecoupling,
+            SmartDecoupling
+        }
+
+        public bool LeftController
+        {
+            get { return leftController; }
+            set
+            {
+                leftController = value;
+                SetControllerListeners(controllerLeftHand);
+            }
+        }
+
+        public bool RightController
+        {
+            get { return rightController; }
+            set
+            {
+                rightController = value;
+                SetControllerListeners(controllerRightHand);
+            }
+        }
+
+        [Tooltip("If this is checked then the left controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool leftController = true;
+        [Tooltip("If this is checked then the right controller touchpad will be enabled to move the play area. It can also be toggled at runtime.")]
+        [SerializeField]
+        private bool rightController = true;
+
+        [Tooltip("Select which trackables are used to determine movement.")]
+        [SerializeField]
+        private ControlOptions m_controlOptions = ControlOptions.HeadsetAndControllers;
+
+        [SerializeField]
+        [Tooltip("Lower to decrease speed, raise to increase.")]
+        private float m_speedScale = 1;
+
+        [SerializeField]
+        [Tooltip("The max speed you can move in game units. (If 0 or less, max speed is uncapped)")]
+        private float m_maxSpeed = 4;
+
+        [SerializeField]
+        [Tooltip("How your movement direction will be determined.  The Gaze method tends to lead to the least motion sickness.  Smart decoupling is still a Work In Progress.")]
+        private DirectionalMethod m_dirMethod = DirectionalMethod.Gaze;
+
+        [SerializeField]
+        [Tooltip("If we're using Smart Decoupling, all trackables must be within this degree threshold to change direction.")]
+        private float m_smartDecoupleThreshold = 30f;
```

Comment:
Also, why are all these variables private but serialized?

How would you change them at runtime?